### PR TITLE
Add option to install page to customize default branch name

### PR DESCRIPTION
**Pull request by 21h**
Its not good to change default branch name without user opinion. Now Gitea will ask what user want, give user a choice. You know.. free to choice. 

image

### CODE REVIEW

**Comment by 21h** [comment]
yeap, its done. you can type default branch name during first install, but its not cure for already broken gitea installations where default branch suddenly changed to "main".

File: options/locale/locale_ru-RU.ini
Comment (6543) [inline options/locale/locale_ru-RU.ini:176]: will be overwritten ... please add them via crowdin after merge

Comment (21h) [reply options/locale/locale_ru-RU.ini:176]: ok

File: options/locale/locale_en-US.ini
Comment (silverwind) [inline options/locale/locale_en-US.ini:179]:
```suggestion
default_branch_name_helper = Name of the default git branch.
```

Sounds more professional then that other text. Suggesting to leave empty was also not good.

Comment (21h) [reply options/locale/locale_en-US.ini:179]: Absolutely same "leave empty" you can see upper on same page. Its professional, you can see examples of use. More professional do not change this. 

image

**Review by techknowlogick — CHANGES_REQUESTED** [review]
It is currently possible to change this via config already, and I'm hesitant to put yet another option on this page, if anything we should remove some to prevent option overload for the user.

**Comment by 21h** [comment]
> It is currently possible to change this via config already, and I'm hesitant to put yet another option on this page, if anything we should remove some to prevent option overload for the user.

Overload? ) Gitea is not gutenberg editor from Wordpress, 10 options instead of 9 not blow users' brain

**Comment by techknowlogick** [comment]
> we are engineers, not blondies

Gitea is aimed for users from all skill levels, and so by adding a new option it forces a user to think about each and every option.

> option must be set at installation time.

There are hundreds of settings, we can't put them all on this page.

> but its not cure for already broken gitea installations where default branch suddenly changed to "main".

As I mentioned above, it is possible to change this setting via configuration already, and we provide documentation on how to do so.

Thank you for this PR, however as per above I will be closing this now. If you have any questions as to how to change this setting please hop into chat and we can assist you.

**Comment by 21h** [comment]
> Thank you for this PR, however as per above I will be closing this now. If you have any questions as to how to change this setting please hop into chat and we can assist you.

may we ask other maintainer what they think? closing PR just because you think it will be difficult for user isnt professional method. @lunny @zeripath

**Comment by delvh** [comment]
I have to agree with @techknowlogick.
There are way too many config options to allow configuring them all on the install page:
At the moment, our [example config](https://github.com/go-gitea/gitea/blob/main/custom/conf/app.example.ini) has **2400** lines.
I am also in favor of keeping the install process as simple as possible.
This means especially **only displaying the absolute bare minimum to have Gitea running**.
I think this one is minor enough to not be required on the install page, especially as most people don't even want to change it.

**Comment by 21h** [comment]
> I have to agree with @techknowlogick. There are way too many config options to allow configuring them all on the install page: At the moment, our [example config](https://github.com/go-gitea/gitea/blob/main/custom/conf/app.example.ini) has **2400** lines. I am also in favor of keeping the install process as simple as possible. This means especially **only displaying the absolute bare minimum to have Gitea running**. I think this one is minor enough to not be required on the install page, especially as most people don't even want to change it.

no one suggesting to put all parameters on one page, just the default branch name because suddenly changing this parameter can break existing pipelines.

**Comment by delvh** [comment]
I'm curious now:
How can you have an existing pipeline if you haven't set your server up yet?

**Comment by 21h** [comment]
> I'm curious now: How can you have an existing pipeline if you haven't set your server up yet?

as you know app.ini almost empty after install. release ago default branch changed to main and my pipelines was broken. I want to stick it in config file if someone want to change it again, this is a question of gitea stability.

**Comment by delvh** [comment]
https://docs.gitea.io/en-us/config-cheat-sheet/#repository-repository: `DEFAULT_BRANCH`?

**Comment by 21h** [comment]
> https://docs.gitea.io/en-us/config-cheat-sheet/#repository-repository: `DEFAULT_BRANCH`?

I know, this parameter missing in app.ini after install. This PR add it and it will become independent from the next gitea releases.

**Comment by 21h** [comment]
> > https://docs.gitea.io/en-us/config-cheat-sheet/#repository-repository: `DEFAULT_BRANCH`?
> 
> I know, this parameter missing in app.ini after install. This PR add it and it will become independent from the next gitea releases.

Yeah, I don't want gitea to suddenly change the name of a branch just because someone is offended. I want to nail down this setting during installation once and no one will be surprised by red warnings in CI in the morning again.

**Comment by 6543** [comment]
witch warnings in witch CI?

**Comment by 6543** [comment]
instead of adding another option I have to give **tech** right.

we instead should focus on enable config edit via web and add the option to redirect after first install to this page

**Comment by jolheiser** [comment]
This change was communicated in the changelog and [blog post](https://blog.gitea.io/2022/07/gitea-1.17.0-is-released/#-main-as-default-branch-19354httpsgithubcomgo-giteagiteapull19354).

I agree this doesn't really warrant a "crucial" setting needed on the install page. Users who wish to use legacy branch naming can change it in their config as needed.

**Comment by jolheiser** [comment]
Further discussion should likely happen in a new issue or on one of our chat platforms, as it may get lost here on a closed PR. 🙂

**Comment by 21h** [comment]
PR is closed, don't care already.

**Pull request closed without merging**
